cmd/skopeo/inspect.go

```diff
@@ -115,19 +137,33 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 	}
 
 	if opts.config && opts.raw {
-		configBlob, err := img.ConfigBlob(ctx)
-		if err != nil {
-			return fmt.Errorf("Error reading configuration blob: %v", err)
+		var configBlob []byte
+		if err := retryIfNecessary(ctx, func() error {
+			blob, err := img.ConfigBlob(ctx)
+			configBlob = blob
+			if err != nil {
+				return errors.Wrapf(err, "Error reading configuration blob")
+			}
+			return nil
+		}, opts.retryOpts, opts.global.commandTimeout); err != nil {
+			return err
 		}
```

Comment:
A very tentative thought: Would it make sense to integrate the very common `errors.Wrapf` into `retryIfNecessary`, something like this:
```go
var configBlob []byte
if err := opts.retryOpts.retryIfNecessary(ctx, func() error {
    configBlob, err = img.ConfigBlob(ctx)
    return err
}, "Error reading configuration blob", /* Other format parameters for Wrapf, if any */); err != nil {
    return err
}
```

That would (along with not using `:=`) let us go from the original 4 lines to 7 instead of 11, OTOH it is also much more magic. Is that worth it or not?

Reply:
if the retry only wraps one function, there will only one kind of error message to return, can this error be wrapped after calling `retryIfNecessary`?
```go
var configBlob []byte
if err := opts.retryOpts.retryIfNecessary(ctx, func() error {
    configBlob, err = img.ConfigBlob(ctx)
    return err
}; err != nil {
    return errors.Wrapf(err,  "Error reading configuration blob")
}
```

Reply:
Oh, right. That’s much clearer.